packages/reakit/src/Portal/Portal.tsx

```diff
@@ -15,7 +15,7 @@ export const PortalContext = React.createContext<HTMLElement | null>(
 export function Portal({ children }: PortalProps) {
   // if it's a nested portal, context is the parent portal
   // otherwise it's document.body
-  const context = React.useContext(PortalContext);
+  const context = React.useContext(PortalContext) || document.body;
```

Comment:
`document` will be `undefined` when this component is rendered on the server.
```suggestion
  const body = typeof document !== "undefined" ? document.body : null;
  const context = React.useContext(PortalContext) || body;
```

Reply:
https://github.com/reakit/reakit/pull/514/commits/aab7580e9690f5611fe0d5956e2707cac2bc06e5 fine like this?